app/controllers/CommentController.php

```diff
@@ -27,109 +28,152 @@ public function addComment($cocktailId)
         $parentCommentId = isset($_POST['parent_comment_id']) ? sanitize($_POST['parent_comment_id']) : null;
 
         if (empty($commentText)) {
-            $_SESSION['error'] = 'Comment cannot be empty.';
-            header("Location: /cocktails/{$cocktailId}");
+            http_response_code(400); // Bad Request
+            echo json_encode(['error' => 'Comment cannot be empty.']);
             exit();
         }
 
-        // Ensure parent_comment_id is null for top-level comments
-        $parentCommentId = empty($parentCommentId) ? null : $parentCommentId;
-
-        // Add the comment using service
-        $this->commentService->addComment($userId, $cocktailId, $commentText, $parentCommentId);
-
-        // Redirect back to the cocktail view
-        $cocktailTitle = urlencode($_POST['cocktailTitle']);
-        header("Location: /cocktails/{$cocktailId}-{$cocktailTitle}");
+        try {
+            // Add the comment
+            $this->commentService->addComment($userId, $cocktailId, $commentText, $parentCommentId);
+
+            // Fetch updated cocktail and comments
+            $cocktail = $this->cocktailService->getCocktailById($cocktailId);
+            $comments = $this->commentService->getCommentsWithReplies($cocktailId);
+            $currentUser = (new AuthController())->getCurrentUser(); // Ensure current user is available
+
+            // Render the entire comment section
+            ob_start();
+            require __DIR__ . '/../views/cocktails/comment_section.php';
+            $commentsHtml = ob_get_clean();
+
+            // Return JSON response
+            header('Content-Type: application/json');
+            echo json_encode(['success' => true, 'html' => $commentsHtml]);
+        } catch (Exception $e) {
+            http_response_code(500); // Internal Server Error
+            echo json_encode(['error' => $e->getMessage()]);
```

Comment:
Returning raw error messages, for example ($e->getMessage()) in JSON responses can expose sensitive details about our application. This is a security risk.

Reply:
better approach would be 
```php
catch (Exception $e) {
    // Log the error internally and return a generic message
    error_log($e->getMessage());
    http_response_code(500); // Internal Server Error
    echo json_encode(['error' => 'An unexpected error occurred.']);
}